### PR TITLE
Dev branch

### PR DESCRIPTION
**Pull request by Kristian-Krastev**
fixed some of the logic and added well formated output in the main to test the tasks

### CODE REVIEW

File: 07_array_algos/practice/solutions.cpp [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+  * @file   solutions.cpp
+  * @author Ivan Filipov
+  * @author Kristian Krastev
+  * @date   12.2019
+  * @brief  Solution for function tasks from practice 7.
+  */
+
+#include <iostream>
+#include <random>
+#include <time.h>
```
Comment (IvanFilipov) [inline 07_array_algos/practice/solutions.cpp]: ctime

File: 07_array_algos/practice/solutions.cpp [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+
+const size_t MAX_N = 1024; // max number of elements we can store
+const size_t SORTED_SIZE = 6;
```
Comment (IvanFilipov) [inline 07_array_algos/practice/solutions.cpp]: comments?

File: 07_array_algos/practice/solutions.cpp [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+const size_t MAX_ROWS = 5;
+const size_t MAX_COLS = 5;
+const size_t BOYS_SIZE = 10;
+const size_t GIRLS_SIZE = 10;
+
+// task_01
+bool descending_array(int arr[], size_t size) {
+
+    for (size_t i = 0; i < size - 1; i++) {
+
```
Comment (IvanFilipov) [inline 07_array_algos/practice/solutions.cpp]: no empty line needed

File: 07_array_algos/practice/solutions.cpp (hunk 1 above)
Comment (IvanFilipov) [inline 07_array_algos/practice/solutions.cpp]: `#include <ctime>`

File: 07_array_algos/practice/solutions.cpp [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+    // if empty
+    if (size == 0) {
+        arr[0] = elem;
+        return 1;
+    }
+  
+    // we traverse backwards in order to move the elements one step to the right
+    // to make room for our new element
+    int i;
+    for (i = size - 1; (i >= 0 && arr[i] > elem); i--) {
```
Comment (IvanFilipov) [inline 07_array_algos/practice/solutions.cpp]: `for (int i = (int)size - 1; (i >= 0 && arr[i] > elem); i--)`

File: 07_array_algos/practice/solutions.cpp [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+    arr[i + 1] = elem;
+    return size + 1;
+}
+
+// task_03
+// because we want to return the updated size
+int remove_elem(int arr[], size_t size, int elem) {
+
+    if (size < 1) {
+        std::cout << "There are no elements here :( \n";
```
Comment (IvanFilipov) [inline 07_array_algos/practice/solutions.cpp]: std::endl in stead of '\n'

File: 07_array_algos/practice/solutions.cpp [hunk 6]
```diff
@@ -0,0 +1,357 @@
...
+    return size - 1;
+}
+
+// task_04
+bool is_symetric(int arr[], size_t size) {
+
+    size_t middle = size / 2;
+    size_t i = 0;
+
+    while ((i < middle) && (arr[i] == arr[size - 1 - i])) {
```
Comment (IvanFilipov) [inline 07_array_algos/practice/solutions.cpp]: `for (size_t i = 0; i < middle; i++)
     if (arr[i] != arr[size - 1 - i])
        return false;
`
....

`return true;`

File: 07_array_algos/practice/solutions.cpp [hunk 7]
```diff
@@ -0,0 +1,357 @@
...
+}
+
+// helper function to sort the array in task_05
+void sort_arr(int arr[], size_t size) {
+
+    int min_val;
+    int min_ind;
+
+    for (size_t i = 0; i < size - 1; i++) {
+
```
Comment (IvanFilipov) [inline 07_array_algos/practice/solutions.cpp:105]: no empty line here

File: 07_array_algos/practice/solutions.cpp [hunk 8]
```diff
@@ -0,0 +1,357 @@
...
+        min_ind = i;
+
+        for (size_t j = i + 1; j < size; j++) {
+            if (arr[j] < min_val) {
+                min_val = arr[j];
+                min_ind = j;
+            }
+        }
+        
+        if (min_ind != (int)i) {
```
Comment (IvanFilipov) [inline 07_array_algos/practice/solutions.cpp]: in the code below, std::swap is used, why not using it here also?

File: 07_array_algos/practice/solutions.cpp [hunk 9]
```diff
@@ -0,0 +1,357 @@
...
+        }
+    }
+}
+
+// task_05
+int filter_primes(int src_arr[], int dest_arr[], size_t size) {
+
+    int counter = 0;
+    for (size_t i = 0; i < size; i++) {
+
```
Comment (IvanFilipov) [inline 07_array_algos/practice/solutions.cpp]: -

File: 07_array_algos/practice/solutions.cpp [hunk 10]
```diff
@@ -0,0 +1,357 @@
...
+void unite_arrays(int first_arr[], int second_arr[], int result_arr[], size_t size) {
+
+    size_t counter = 0;
+    size_t i = 0;
+    size_t j = 0;
+
+    while (i < size && j < size) {
+        if (first_arr[i] <= second_arr[j])
+            result_arr[counter++] = first_arr[i++];
+
```
Comment (IvanFilipov) [inline 07_array_algos/practice/solutions.cpp]: ???

File: 07_array_algos/practice/solutions.cpp [hunk 11]
```diff
@@ -0,0 +1,357 @@
...
+    while(i < size)
+        result_arr[counter++] = first_arr[i++];
+
+    while (j < size) 
+        result_arr[counter++] = second_arr[j++];
+}
+
+// task_07
+void sort_matrix(int matrix[][MAX_COLS], size_t rows, size_t cols) {
+    for (size_t i = 0; i < rows; i++) {
```
Comment (IvanFilipov) [inline 07_array_algos/practice/solutions.cpp]: empty line here is a necessary

File: 07_array_algos/practice/solutions.cpp [hunk 12]
```diff
@@ -0,0 +1,357 @@
...
+
+    return min_elem;
+}
+
+void print_array(int arr[], size_t size) {
+
+    for (size_t i = 0; i < size; i++) {
+        std::cout << arr[i] << ' ';
+    }
+    std::cout << '\n';
```
Comment (IvanFilipov) [inline 07_array_algos/practice/solutions.cpp]: std::endl

File: 07_array_algos/practice/solutions.cpp [hunk 13]
```diff
@@ -0,0 +1,357 @@
...
+
+		// get the two smallest people and 'remove' them so
+		// that they dont take part in the next minimum
+        int smallest_boy = get_min(boys, i);
+        int smallest_girl = get_min(girls, i);
+        result[cnt++] = smallest_boy;
+        result[cnt++] = smallest_girl;
+    }
+
+	std::cout << "\nComibned array: \n";
```
Comment (IvanFilipov) [inline 07_array_algos/practice/solutions.cpp]: indentation

**Review by IvanFilipov — COMMENTED** [review]
General comments:
- the indentations are inconsistent, maybe your editor have messed them.
- stick to std::endl, not to '\n'

File: 07_array_algos/practice/solutions.cpp (hunk 2 above)
Comment (IvanFilipov) [reply 07_array_algos/practice/solutions.cpp]: and `include <ctime>`

File: 07_array_algos/practice/solutions.cpp (hunk 10 above)
Comment (IvanFilipov) [reply 07_array_algos/practice/solutions.cpp]: a strange empty line here

**Comment by IvanFilipov** [comment]
Also, you can try to write a little bit more verbose commit messages. 
For example:

> adding solutions for tasks pr 7

is quite OK, but

> update solutions

 and 

> final version of solutions.cpp

are confusing, because we have many files called `solutions.cpp` .

File: 07_array_algos/practice/solutions.cpp [hunk 14]
```diff
@@ -0,0 +1,357 @@
...
+	{{ 5, 8, 12, 95, 42 },
+	 { 3, 9, 7, 1, 6 },
+	 { 76, 4, 233, 63, 369},
+	 { 93, 943, 73, 1, 2 },
+	 { 17, 41, 76, 34, 45 } };
+
+	std::cout << "Matrix before it is sorted: \n";
+	print_matrix(matrix, MAX_ROWS);
+
+	sort_matrix(matrix, MAX_ROWS, MAX_COLS);
```
Comment (IvanFilipov) [inline 07_array_algos/practice/solutions.cpp:343]: `cout << "Matrix after sorting` or smth

File: 07_array_algos/practice/solutions.cpp [hunk 15]
```diff
@@ -0,0 +1,357 @@
...
+        result[cnt++] = smallest_boy;
+        result[cnt++] = smallest_girl;
+    }
+
+	std::cout << "\nComibned array: \n";
+    print_array(result, BOYS_SIZE + GIRLS_SIZE);
+}
+
+void print_matrix(int matrix[][MAX_COLS], size_t rows)
+{
```
Comment (IvanFilipov) [inline 07_array_algos/practice/solutions.cpp]: where is the bracket? And the output can be formatted a just a little bit better. We've got code to do so. The idea is the use std::setw

File: 07_array_algos/practice/solutions.cpp [hunk 16]
```diff
@@ -0,0 +1,357 @@
...
+	int first_arr[SORTED_SIZE] = { 10, 12, 15, 20, 31, 81 };
+	int second_arr[SORTED_SIZE] = { 1, 7, 18, 20, 26, 102 };
+	int third_arr[2* SORTED_SIZE];
+	unite_arrays(first_arr, second_arr, third_arr, SORTED_SIZE);
+	std::cout << "First array + second array in sorted way: \n";
+	print_array(third_arr, 2 * SORTED_SIZE);
+
+	std::cout << "\n-----------Task 7-----------\n";
+	int matrix[MAX_ROWS][MAX_COLS] =
+	{{ 5, 8, 12, 95, 42 },
```
Comment (IvanFilipov) [inline 07_array_algos/practice/solutions.cpp]: the format of the matrix's initialization list differs from the other examples.

File: 07_array_algos/practice/solutions.cpp (hunk 9 above)
Comment (Kristian-Krastev) [reply 07_array_algos/practice/solutions.cpp]: what does that dot mean

Comment (IvanFilipov) [reply 07_array_algos/practice/solutions.cpp]: it was actually a dash "-", with the meaning of "the same mistake as described above".

**Pull request merged**